apps/emqx_gcp_device/src/emqx_gcp_device_api.erl

```diff
@@ -0,0 +1,479 @@
+                ]
+            }
+        },
+        post => #{
+            description => ?DESC(gcp_devices_post),
+            tags => ?TAGS,
+            'requestBody' => hoconsc:mk(hoconsc:array(?R_REF(gcp_exported_device)), #{}),
+            responses =>
+                #{
+                    200 => hoconsc:mk(hoconsc:ref(import_result)),
+                    400 => emqx_dashboard_swagger:error_codes(
+                        ['BAD_REQUEST'],
+                        <<"Bad Request">>
+                    )
+                }
+        }
+    };
+schema("/gcp_devices/:deviceid") ->
+    #{
+        'operationId' => '/gcp_devices/:deviceid',
+        get =>
+            #{
+                description => ?DESC(gcp_device_get),
+                tags => ?TAGS,
+                parameters => [
+                    {deviceid,
+                        hoconsc:mk(binary(), #{
+                            desc => ?DESC(deviceid),
+                            required => true,
+                            in => path,
+                            example => <<"c2-ec-x509">>
+                        })}
+                ],
+                responses =>
+                    #{
+                        200 => hoconsc:mk(
+                            hoconsc:ref(gcp_device_all_info),
+                            #{
+                                desc => ?DESC(gcp_device_all_info)
+                            }
+                        ),
+                        404 => emqx_dashboard_swagger:error_codes(
+                            ['NOT_FOUND'],
+                            ?DESC(gcp_device_response404)
+                        )
+                    }
+            },
+        put =>
+            #{
+                description => ?DESC(gcp_device_put),
+                tags => ?TAGS,
+                parameters => [
+                    {deviceid,
+                        hoconsc:mk(binary(), #{
+                            desc => ?DESC(deviceid),
+                            required => true,
+                            in => path,
+                            example => <<"c2-ec-x509">>
+                        })}
+                ],
+                'requestBody' => hoconsc:ref(gcp_device),
+                responses =>
+                    #{
+                        200 => hoconsc:mk(
+                            hoconsc:ref(gcp_device_info),
+                            #{
+                                desc => ?DESC(gcp_device_info)
+                            }
+                        ),
+                        400 => emqx_dashboard_swagger:error_codes(
+                            ['BAD_REQUEST'],
+                            <<"Bad Request">>
+                        )
+                    }
+            },
+        delete => #{
+            description => ?DESC(gcp_device_delete),
+            tags => ?TAGS,
+            parameters => [
+                {deviceid,
+                    hoconsc:mk(binary(), #{
```

Comment:
Nit: since the same path param is used multiple times, it could be moved to a fn definition and shared.